### DISCUSSION TITLE
Private indicators

### DISCUSSION THREAD

**Discussion by jppeltier** [Mathieu2301/TradingView-API#36] [Q&A] [answered]
Hello Mathieu, and congratulations for the work done so far!

I have private indicators, that I can obviously access if and only if I am personally connected to the platform. I checked through your code, but could not find any way to input login/password and get a private session.

Hence my question: is there any way to use your tools as a connected user (preferably by generating a private session from login/password, or alternatively by opening a private session in a browser and passing by some relevant key to your tools)?

Thanks,

**Comment by Mathieu2301** [comment, marked as answer]
On V3, we have the `token` parameter for `Client` object.
```javascript
const client = new TradingView.Client({
  token: 'YOUR_SESSIONID_COOKIE',
});
```

**Reply by jppeltier** [reply]
Thanks, looking forward to V3 then!

**Reply by Mathieu2301** [reply]
The V3 is already available !
You can download it by selecting [the `v3.0-remake` branch](https://github.com/Mathieu2301/TradingView-API/tree/v3.0-remake).
Or maybe with the `npm i github:Mathieu2301/TradingView-API#v3.0-remake` command.

**Reply by MouloudSaidoun** [reply]
Hello Mathieu,

Thank you for this great work.
I am trying to use private indicators and I am wondering how to do that.
if we take this example:
`const client = new TradingView.Client({
  token: 'YOUR_SESSIONID_COOKIE',
});`

How to get the 'SESSIONID_COOKIE' ? 
is there any other options to connect to the private TV ?

Thanks

**Reply by Mathieu2301** [reply]
Hello, you have to check in your cookies.

**Reply by Braxt0nAI** [reply]
Hi, @Mathieu2301 quick question about invite-only / private indicators.

Do you consider the two to be the same thing? I have a few paid indicators that I am trying to access using my session cookie and keep getting an error **"study_not_auth:Script@tv-scripting-332 undefined"**. Do I need to pass additional data in order to access paid invite-only indicators? Thank you in advance!

**Reply by Mathieu2301** [reply]
> Do you consider the two to be the same thing?

We consider that an indicator is private when we need to login to access it.
 
> I have a few paid indicators that I am trying to access using my session cookie and keep getting an error "study_not_auth:Script@tv-scripting-332 undefined". Do I need to pass additional data in order to access paid invite-only indicators? Thank you in advance!

Weird... Make sure you have entered the correct session token and pineID and it should work (at least if it works on tradingview).

**Reply by Braxt0nAI** [reply]
@Mathieu2301 Thank you for the quick response. Yes, I have checked the session ID and indicator ID to ensure they were correct, along with testing other indicators in my account that were not invite-only/private and they seem to work just fine. 

Because it's paid, do I need to add the code from examples/userlogin.js to include the username & password along with the session ID to authenticate?

**Reply by Mathieu2301** [reply]
> Because it's paid, do I need to add the code from examples/userlogin.js to include the username & password along with the session ID to authenticate?

Nope, `login` function is only an alternative way to get a `sessionid`.
